### PR TITLE
SPLICE-2108 Can't disable logging statements

### PR DESCRIPTION
**Pull request by OlegMazurov**
A new fix checking for system property "derby.language.logStatementText", config property "splice.debug.logStatementContext" and passing them down via startParams to PropertyConglomerate.

### CODE REVIEW

**Comment by wb14123** [comment]
+1

**Comment by wtang16** [comment]
@OlegMazurov Can you update JIRA with latest discussion/conclusion? Also, if documentation needs to be modified, please point out.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java [hunk 1]
```diff
@@ -77,23 +77,34 @@
...
             System.setProperty("derby.drda.timeSlice","60000");
 
         //  System.setProperty("derby.language.logQueryPlan", Boolean.toString(true));
-        if(config.debugLogStatementContext()) {
-            System.setProperty("com.splicemachine.enableLegacyAsserts",Boolean.TRUE.toString());
-            System.setProperty("derby.language.logStatementText",Boolean.toString(true));
+        String logStatementText = System.getProperty("derby.language.logStatementText");
+        if (logStatementText == null) {
+            if (config.debugLogStatementContext()) {
+                System.setProperty("com.splicemachine.enableLegacyAsserts", Boolean.TRUE.toString());
```
[the remark this answers is not in the document]
Comment (OlegMazurov) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java]: Indeed. That enables DEBUG and ASSERT checks in SanityManager, which we didn't have on in production. I think we should either remove setting "com.splicemachine.enableLegacyAsserts" from here or set it if the "derby.language.logStatementText" system property is true.
Please "request changes" for this PR.

[the remark this answers is not in the document]
Comment (OlegMazurov) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/db/SpliceDatabase.java]: I have removed "com.splicemachine.enableLegacyAsserts". Should be set explicitly on the command line if needed.

**Pull request merged**